lib/src/source/git.dart

```diff
@@ -426,6 +433,22 @@ class BoundGitSource extends CachedSource {
     _updatedRepos.add(path);
   }
 
+  /// Check if [directory] is a valid git repo. If invalid and
+  /// [delIfInvalid] is set to be true, it will be deleted.
+  ///
+  bool _checkIfValidGitRepo(Directory directory, {bool delIfInvalid = false}) {
+    if (directory.existsSync()) {
+      var result = runProcessSync(git.command, ['log', '-1'],
```

Comment:
If we're just checking if this is a valid git repository why not use: `git rev-parse --is-inside-work-tree`

Is `git log` better?

I think `git rev-parse --is-inside-work-tree` was introduced in:
https://github.com/git/git/commit/892c41b98ae2e6baf3aa13901cb10db9ac67d2f3
Which shipped with git 1.5.3 in 2007, so it's probably fair to depend upon this today.

Reply:
'--is-inside-git-dir' would be the right one to check a git repo cache.
It's introduced in https://github.com/git/git/commit/c9bf7be238ac3c767f9ac6546d5bb7b11765253c 
on Jun 17, 2007